Sources/SPTDataLoader/SPTDataLoaderService.m

```diff
@@ -405,16 +500,21 @@ - (void)URLSession:(NSURLSession *)session
         } else {
             handler.task = [session dataTaskWithRequest:handler.request.urlRequest];
         }
+    } else {
+        [_crashReporter leaveBreadcrumb:
+         [NSString stringWithFormat:@"Trying to retry on invalidated sesion. Policy: %ld, Error: %@",
+          (long)handler.request.backgroundPolicy, error.description]];
     }
```

Comment:
I am not sure about this one. The retry is an internal feature of SPTDataLoader. The `if` at line 402/497 was added to fix a bug where SPTDataLoader was trying to retry on an already invalidated sessions. And to add more on it, entering the `else` part does not guarantee the session was invalidated, it can just be that single task got cancelled and thus the message could be mis-leading as well

Reply:
Since this code can't retry on invalidated session anymore, I'll just remove this log then